src/main/java/com/aws/greengrass/componentmanager/plugins/docker/DockerImageDownloader.java

```diff
@@ -276,4 +286,50 @@ private <T> T runWithConnectionErrorCheck(CrashableSupplier<T, Exception> task)
             throw e;
         }
     }
+
+    /**
+     * Cleanup component, delete docker image when component being removed.
+     * @throws IOException exception
+     */
+    @Override
+    public void cleanup() throws IOException {
+        // this docker image not only used by itself
+        try {
+            if (!ifImageUsedByOther(componentStore)) {
+                Image image = DockerImageArtifactParser
+                        .getImage(ComponentArtifact.builder().artifactUri(artifact.getArtifactUri()).build());
+                dockerClient.deleteImage(image);
+            }
+        } catch (PackageLoadingException | InvalidArtifactUriException | DockerImageDeleteException e) {
+            throw new IOException(e);
+        }
+    }
+
+    /**
+     * @param componentStore componentStore
+     * @return true: this image used by other; false: not used.
+     * @throws PackageLoadingException from getPackageRecipe
+     */
+    public boolean ifImageUsedByOther(ComponentStore componentStore) throws PackageLoadingException {
+        Map<String, Set<String>> allVersions = componentStore.listAvailableComponentVersions();
+        for (Map.Entry<String, Set<String>> versions : allVersions.entrySet()) {
+            String compName = versions.getKey();
+            Set<String> localVersions = new HashSet<>(versions.getValue());
+            for (String compVersion : localVersions) {
+                try {
+                    ComponentIdentifier identifier = new ComponentIdentifier(compName, new Semver(compVersion));
+                    if (ObjectUtils.notEqual(identifier, this.identifier)) {
```

Comment:
Do not use `ObjectUtils` to compare since already have `compareTo` defined for ComponentIdentifier